here-naksha-lib-psql/src/test/java/com/here/naksha/lib/psql/PsqlStorageTests.java

```diff
@@ -348,6 +350,51 @@ void testDuplicateFeatureId() throws NoCursor {
     }
   }
 
+  @Test
+  @Order(61)
+  @EnabledIf("runTest")
+  void testMultiOperationPartialFail() throws NoCursor {
+    assertNotNull(storage);
+    assertNotNull(session);
+
+    final String UNIQUE_VIOLATION_CODE = "23505";
+    final String UNIQUE_VIOLATION_MSG =
+        format("The feature with the id '%s' does exist already", SINGLE_FEATURE_ID);
+
+    // given
+    final WriteXyzFeatures request = new WriteXyzFeatures(collectionId());
+    final XyzFeature featureToSucceed = new XyzFeature("123");
+    request.add(EWriteOp.CREATE, featureToSucceed);
+    final XyzFeature featureToFail = new XyzFeature(SINGLE_FEATURE_ID);
+    request.add(EWriteOp.CREATE, featureToFail);
+
+    // when
+    final Result result = session.execute(request);
+
+    // then
+    assertInstanceOf(ErrorResult.class, result);
+    ErrorResult errorResult = (ErrorResult) result;
+    assertEquals(UNIQUE_VIOLATION_CODE, errorResult.reason.value());
+    assertEquals(UNIQUE_VIOLATION_MSG, errorResult.message);
+    try (ForwardCursor<XyzFeature, XyzFeatureCodec> cursor = result.getXyzFeatureCursor()) {
+      assertTrue(cursor.next());
+      assertEquals(featureToSucceed.getId(), cursor.getFeature().getId());
+
+      // error row check
+      assertTrue(cursor.next());
+      assertTrue(cursor.hasError());
+      assertSame(EExecutedOp.ERROR, cursor.getOp());
+      CodecError rowError = cursor.getError();
+      assertNotNull(rowError);
+      assertEquals(UNIQUE_VIOLATION_CODE, rowError.err.value());
```

Comment:
Based on how @xeus2001 explained, this was expected to return XyzErr.CONFLICT . If we have now decided to return actual Postgres error back, that is also fine as we haven't started with psql alignment yet. But we need to ensure there is consistent agreed error handling going forward (and so the next question will be, how will no-data-found error will be returned for Update scenario?)

Reply:
I agree to this, but maybe we can make this a two step fix. So lets first approve this change, as it improves the situation and then ensure that we somehow translate the unique violation code into a CONFLICT.

Reply:
Ok, so we then continue with XyzErr specific codes. Yes, it makes sense not to hold this PR, as we haven't started with psql alignment yet, so no impact.